auth.go

```diff
@@ -426,77 +456,120 @@ func (x *Central) Logout(sessionkey string) error {
 	}
-	x.Log.Infof("ResetPasswordStart successful (%v)", identity)
+	x.Log.Infof("ResetPasswordStart successful (%v)", userId)
 	return token, err
 }
 
 // Complete the password reset process, by providing a token that was generated by ResetPasswordStart.
 // If this succeeds, then the password is set to 'password', and the token becomes invalid.
-func (x *Central) ResetPasswordFinish(identity string, token string, password string) error {
-	if err := x.authenticator.ResetPasswordFinish(identity, token, password); err != nil {
-		x.Log.Infof("ResetPasswordFinish failed (%v) (%v)", identity, err)
+func (x *Central) ResetPasswordFinish(userId UserId, token string, password string) error {
+	if err := x.userStore.ResetPasswordFinish(userId, token, password); err != nil {
+		x.Log.Infof("ResetPasswordFinish failed (%v) (%v)", userId, err)
 		return err
 	}
-	x.Log.Infof("ResetPasswordFinish successful (%v)", identity)
-	x.sessionDB.InvalidateSessionsForIdentity(identity)
+	x.Log.Infof("ResetPasswordFinish successful (%v)", userId)
+	x.sessionDB.InvalidateSessionsForIdentity(userId)
 	return nil
 }
 
-// Create an identity in the Authenticator.
-// For the equivalent operation in the PermitDB, simply call SetPermit()
-func (x *Central) CreateAuthenticatorIdentity(identity, password string) error {
-	e := x.authenticator.CreateIdentity(identity, password)
+// Create an identity in the AuthUserStore.
+func (x *Central) CreateUserStoreIdentity(email, username, firstname, lastname, mobilenumber, password string) error {
+	e := x.userStore.CreateIdentity(email, username, firstname, lastname, mobilenumber, password)
+	if e == nil {
+		x.Log.Infof("CreateAuthenticatorIdentity successful (%v)", email)
+	} else {
+		x.Log.Infof("CreateAuthenticatorIdentity failed (%v) (%v)", email, e)
+	}
+	return e
+}
+
+// Update a user in the AuthUserStore.
+func (x *Central) UpdateIdentity(userId UserId, email, username, firstname, lastname, mobilenumber string) error {
+	e := x.userStore.UpdateIdentity(userId, email, username, firstname, lastname, mobilenumber)
 	if e == nil {
-		x.Log.Infof("CreateAuthenticatorIdentity successful (%v)", identity)
+		x.Log.Infof("Update Identity successful (%v)", userId)
 	} else {
-		x.Log.Infof("CreateAuthenticatorIdentity failed (%v) (%v)", identity, e)
+		x.Log.Infof("Update Identity failed (%v) (%v)", userId, e)
 	}
 	return e
 }
 
+// Archive a user in the AuthUserStore.
+func (x *Central) ArchiveIdentity(userId UserId) error {
+	e := x.userStore.ArchiveIdentity(userId)
+	if e == nil {
+		x.Log.Infof("Archive Identity successful (%v)", userId)
+	} else {
+		x.Log.Infof("Archive Identity failed (%v) (%v)", userId, e)
+	}
+	return e
+}
+
+// Get userid from identity.
+func (x *Central) GetUserIdFromIdentity(identity string) (UserId, error) {
+	userId, e := x.userStore.GetUserIdFromIdentity(identity)
+	if e == nil {
+		return userId, nil
+	} else {
+		x.Log.Infof("GetUserIdFromIdentity failed (%v) (%v)", identity, e)
+	}
+	return -1, e
```

Comment:
We need a constant NullUserId, which we should return here:
`const NullUserId UserId = 0`
Unless you have constraints which dictates otherwise, you should always prefer zero over -1 for a null value. The reason is that 0 is the default value for an integer in Go. The same applies to C, because there you want to be able to use memset(0) to initialize a data structure. Also, I have never met a database that will return 0 as it's first autogenerated ID, so 0 should be the de-facto null value for integers.